distributed/scheduler.py

```diff
@@ -6224,7 +6224,11 @@ async def feed(
             if teardown:
                 teardown(self, state)
 
-    def log_worker_event(self, worker=None, topic=None, msg=None):
+    def log_worker_event(
+        self, worker: str, topic: str | Collection[str], msg: Any
+    ) -> None:
+        if isinstance(msg, dict):
+            msg["worker"] = worker
```

Comment:
I'm curious, why this change?

For context, this tripped me up when doing some Coiled work.  I'm not against it, I'm just curious about the motivation.

Reply:
Previously, the same info was manually added to *most* worker-side calls to log-event. Adding it when it lands looks cleaner and less redundant.

Reply:
Users can use this system too though.  We're adding this to their messages.

Reply:
Also, when do you sleep?  I'm about to finish up my work day in California.  If you're working past when I do then that really puts me to shame :)

Reply:
(travelling away from home in Texas this week)